pkg/util/containerd/namespaces_test.go

```diff
@@ -68,16 +68,17 @@ func TestNamespacesToWatch(t *testing.T) {
 		},
 	}
 
-	originalContainerdNamespacesOpt := pkgconfigsetup.Datadog().GetStringSlice("containerd_namespaces")
-	originalExcludeNamespacesOpt := pkgconfigsetup.Datadog().GetStringSlice("containerd_exclude_namespaces")
+	cfg := mock.New(t)
+	originalContainerdNamespacesOpt := cfg.GetStringSlice("containerd_namespaces")
+	originalExcludeNamespacesOpt := cfg.GetStringSlice("containerd_exclude_namespaces")
 
 	for _, test := range tests {
 		t.Run(test.name, func(t *testing.T) {
-			pkgconfigsetup.Datadog().SetWithoutSource("containerd_namespaces", test.containerdNamespaceVal)
-			defer pkgconfigsetup.Datadog().SetWithoutSource("containerd_namespaces", originalContainerdNamespacesOpt)
+			cfg.SetWithoutSource("containerd_namespaces", test.containerdNamespaceVal)
+			defer cfg.SetWithoutSource("containerd_namespaces", originalContainerdNamespacesOpt)
 
-			pkgconfigsetup.Datadog().SetWithoutSource("containerd_exclude_namespaces", test.excludeNamespaceVal)
-			defer pkgconfigsetup.Datadog().SetWithoutSource("containerd_exclude_namespaces", originalExcludeNamespacesOpt)
+			cfg.SetWithoutSource("containerd_exclude_namespaces", test.excludeNamespaceVal)
+			defer cfg.SetWithoutSource("containerd_exclude_namespaces", originalExcludeNamespacesOpt)
```

Comment:
Same remark here: if the `cfg` mock config has a lifetime limited to the execution of the test, do we still need to restore previous config parameters value in `defer` statements?

Reply:
Same here, removed the defer.